### PR TITLE
Storybook: added ability to mark pages with their status regarding quality

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

Add ability to mark every page with status regarding quality, using one of 3 predefined values: `bad`, `decent`, `good`, described in `CONTRIBUTING.md`.

Marking works in the same way as for categories and figma links. What is needed is a comment is a page file, e.g.:
> // status: good

Moreover, there is dialog added providing basic statistics regarding statuses.

To do next: initial categorization of existing pages

Closes: #16606

[Screencast from 30.10.2024 12:28:35.webm](https://github.com/user-attachments/assets/dda49c91-d984-423d-9d0f-25639ce01746)

### Affected areas
`Storybook`

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -381,6 +381,26 @@ It offers a number of functionalities that improve development:
   relative import from the file system. As a consequence, the mechanism for
   overriding import paths for tests and Storybook's pages will not work.
 
+### Page classification
+
+To keep the component and Storybook pages in the good shape, a simple classification
```
Comment (jrainville) [inline CONTRIBUTING.md:386]:
```suggestion
To keep the components and Storybook pages in a good shape, a simple classification
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -381,6 +381,26 @@ It offers a number of functionalities that improve development:
...
+has been introduced. It allows to assign a page one of the following statuses:
```
Comment (jrainville) [inline CONTRIBUTING.md:387]:
```suggestion
has been introduced. It allows to assign pages one of the following statuses:
```

**Review by jrainville — APPROVED** [review]
Looks good, just small typos

File: storybook/pagesmodel.h [hunk 3]
```diff
@@ -12,6 +12,7 @@ struct PagesModelItem {
     QDateTime lastModified;
     QString title;
     QString category;
+    int status = 0;
```
[the remark this answers is not in the document]
Comment (micieslak) [reply storybook/pagesmodel.h:15]: Rather not possible because there is no way to forward declare nested enum :thinking:

**Review by alexjba — APPROVED** [review]
Nice! I'm wondering if it's worth considering the qml tests in the quality status since it's something integrated in SB

File: storybook/pagesmodel.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (caybro) [reply storybook/pagesmodel.h:15]: Right, you have to put the `struct` after the `class`? :)

**Pull request merged**